### PR TITLE
Added the no-progress flag for project.py status to hide the progress…

### PR DESCRIPTION
**Pull request by iblanco11981870**
## Description

Added the no-progress flag for `FlowProject.print_status`. This hides the progress bar output generated when `FlowProject.print_status`, and adds the options to hide the progress bar when desired.
## Motivation and Context

The progress bar displays a significant amount of the output for small workspaces and does not give any benefit. Additionally, using `FlowProject.print_status` in Jupyter notebooks requires special configurations or else it does not work. This addresses issue #602 
## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-flow/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: flow/util/misc.py [hunk 1]
```diff
@@ -359,37 +362,59 @@ def _get_parallel_executor(parallelization="none"):
...
-        def parallel_executor(func, iterable, **kwargs):
-            if "chunksize" in kwargs:
-                # Chunk size only applies to thread/process parallel executors
-                del kwargs["chunksize"]
-            return list(tmap(func, iterable, tqdm_class=tqdm, **kwargs))
+            def parallel_executor(func, iterable, **kwargs):
+                result = []
+                for i in iterable:
+                    result.append(func(i))
+                return result
```
Comment (b-butler) [inline flow/util/misc.py:417]:
```suggestion
                return [func(i) for i in iterable]
```

File: flow/util/misc.py [hunk 2]
```diff
@@ -359,37 +362,59 @@ def _get_parallel_executor(parallelization="none"):
 
     """
     if parallelization == "thread":
```
Comment (b-butler) [inline flow/util/misc.py:364]: This would be cleaner as a nested if with `progress` first then `parallelization`.

File: flow/util/misc.py [hunk 3]
```diff
@@ -359,37 +362,59 @@ def _get_parallel_executor(parallelization="none"):
...
+                    partial(_run_cloudpickled_func, cloudpickle.dumps(func)),
+                    map(cloudpickle.dumps, iterable),
+                    tqdm_class=tqdm,
+                    **kwargs,
+                )
+
+        else:
+
+            def parallel_executor(func, iterable, **kwargs):
+                return ProcessPoolExecutor().map(func, iterable)
```
Comment (b-butler) [inline flow/util/misc.py:400]: We should test the `progress` flag with process and thread parallelization.

**Review by b-butler — CHANGES_REQUESTED** [review]

File: flow/project.py [hunk 4]
```diff
@@ -5006,7 +5014,10 @@ class MyProject(FlowProject):
             "for the majority of runtime required for status determination. "
             "Optionally provide a filename pattern to select for what files "
             "to show result for. Defaults to the main module. "
-            "(requires pprofile)",
+            "(requires profile)",
```
Comment (bdice) [inline flow/project.py:5017]: This is not a typo. The package we import is called `pprofile`. https://github.com/glotzerlab/signac-flow/blob/38a7b8b425cf4a1a1d4faa73a3e1b8d2dc6573e3/flow/project.py#L2904

```suggestion
            "(requires pprofile)",
```

File: tests/test_status.py [hunk 5]
```diff
@@ -6,9 +6,41 @@
...
+        for job in status_pr:
+            kwargs = job.statepoint()
+            tmp_err = io.TextIOWrapper(io.BytesIO(), sys.stderr.encoding)
+            fp.print_status(**kwargs, err=tmp_err, no_progress=hide_progress_bar)
+            tmp_err.seek(0)
+            generated_tqdm = tmp_err.read()
+            if hide_progress_bar:
+                assert "Fetching status" not in generated_tqdm
+            else:
+                assert "Fetching status" in generated_tqdm
```
Comment (bdice) [inline tests/test_status.py:41]:
```suggestion
            assert ("Fetching status" not in generated_tqdm) == hide_progress_bar
```

**Review by bdice — CHANGES_REQUESTED** [review]
Thanks for working on this @iblanco11981870! I had to rebase this PR because it contained some merged commits that were included by accident and made the PR diff appear incorrectly. Please run these commands to fix your local branch:
```bash
$ git fetch --all
$ git checkout feature/print_status
$ git reset --hard origin/feature/print_status
```

I have some design input as well. See comments. Thank you!

**Comment by joaander** [comment]
What more needs to be done to complete this pull request? I would like to use it to clean up the output in the HOOMD-blue tutorials.

**Comment by klywang** [comment]
@joaander I think we're just waiting on a reviewer. I addressed all the suggested changes.

**Comment by joaander** [comment]
OK, the reviewers may not know unless you click the "re-request review" button or otherwise communicate that to them.

**Review by b-butler — APPROVED** [review]
Thanks for the work @iblanco11981870 and @klywang!

**Comment by iblanco11981870** [comment]
@bdice @Tobias-Dwyer We addressed the changes requested. Could you please take a look at it when you have time?

File: changelog.txt [hunk 6]
```diff
@@ -15,6 +15,7 @@ Added
 +++++
 
 - Added the OCLF Crusher environment (#708).
+- Added feature to hide the status of the progress bar (#685).
```
Comment (bdice) [inline changelog.txt:18]: This needs to be moved to a new version entry, since 0.24.0 has already been released.

**Review by bdice — CHANGES_REQUESTED** [review]
https://github.com/glotzerlab/signac-flow/pull/685#discussion_r1039022132

To clarify my previous comment: we should only have "show" or "hide" but not both. Right now there are some functions that accept `progress` and others that accept `hide_progress`. Make this uniform across the functions.

**Comment by bdice** [comment]
@iblanco11981870 Can you please resolve conflicts on this branch before I review?

File: changelog.txt [hunk 7]
```diff
@@ -48,6 +49,7 @@ Changed
 
 - Requires signac 2.0.0 (#734).
 
+
```
Comment (bdice) [inline changelog.txt:52]:
```suggestion
```

File: flow/project.py [hunk 8]
```diff
@@ -3377,8 +3386,8 @@ def _run_operations_in_parallel(self, pool, operations, progress, timeout):
             Process pool.
         operations : Sequence of instances of :class:`_JobOperation`
             The operations to execute.
-        progress : bool
-            Show a progress bar during execution.
+        hide_progress : bool
+            Hides progress bar when printing status output (Default value = True).
```
Comment (bdice) [inline flow/project.py:3390]: There is no default value set here.

File: flow/project.py [hunk 9]
```diff
@@ -5032,9 +5050,9 @@ class MyProject(FlowProject):
             help="Do not actually execute commands, just show them.",
         )
         execution_group.add_argument(
-            "--progress",
-            action="store_true",
-            help="Display a progress bar during execution.",
+            "--hide_progress",
+            action="store_false",
```
Comment (bdice) [inline flow/project.py:5054]: Trying to make sure I understand here -- why is this `store_false` and the other line above is `store_true`? Does the progress behavior match the behavior prior to this PR?

File: tests/test_status.py [hunk 10]
```diff
@@ -6,9 +6,35 @@
...
+def hide_progress_bar(request):
+    return request.param
+
+
+@pytest.fixture(params=["thread", "process", "none"])
+def parallelization(request):
+    return request.param
+
+
+def test_hide_progress_bar(hide_progress_bar, parallelization):
```
Comment (bdice) [inline tests/test_status.py:23]: Do we have a corresponding test for progress bars during execution? That behavior was also changed in this PR.

**Comment by bdice** [comment]
@iblanco11981870 Can we push this through or close it?

File: tests/test_status.py (hunk 10 above)
Comment (b-butler) [reply tests/test_status.py:23]: That part has been removed.

File: flow/project.py (hunk 9 above)
Comment (b-butler) [reply flow/project.py]: This did change behavior and has been removed.

File: flow/project.py (hunk 8 above)
Comment (b-butler) [reply flow/project.py]: Has been reverted.

File: flow/util/misc.py [hunk 11]
```diff
@@ -351,45 +352,70 @@ def _get_parallel_executor(parallelization="none"):
     parallelization : str
         Parallelization mode. Allowed values are "thread", "process", or
         "none". (Default value = "none")
+    hide_progress: bool
```
Comment (bdice) [inline flow/util/misc.py:355]:
```suggestion
    hide_progress : bool
```

**Review by bdice — APPROVED** [review]
@b-butler Thanks for requesting my review on this. I took one brief pass but can't dedicate more time to this PR. Please merge when you think it is sufficiently ready.

**Comment by joaander** [comment]
Thanks!

**Pull request merged**